conda/trust/signature_verification.py

```diff
@@ -227,23 +232,58 @@ def _fetch_channel_signing_data(
                 f"Invalid JSON returned from {signing_data_url}/{filename}"
             )
 
-    def __call__(self, info, fn, signatures):
-        if not self.enabled or fn not in signatures:
+    def verify(self, repodata_fn: str, record: PackageRecord):
+        repodata, _ = SubdirData(
+            record.channel,
+            repodata_fn=repodata_fn,
+        ).repo_fetch.fetch_latest_parsed()
```

Comment:
I added this API. Is this the correct way to hold it, or should conda interact with the higher-level SubdirData? Can we add an "always use the cache/don't download newer repodata than was sent to the solver" argument?

Reply:
This was inspired by what conda-libmamba-solver does [here](https://github.com/conda/conda-libmamba-solver/blob/1443a464480dff908e96f3153d2521a429929637/conda_libmamba_solver/index.py#L232)

happy to create a new helper method in `SubdirData` for this

cc @jaimergp